dashboard/app/controllers/admin_users_controller.rb

```diff
@@ -149,6 +149,35 @@ def delete_progress_form
       count
   end
 
+  def delete_progress
+    params.require([:user_id, :script_id, :reason])
+
+    # TODO: Check for escaping issues for 'reason' once everything is hooked up end-to-end.
+
+    user_id = params[:user_id]
+    script_id = params[:script_id]
+    user_storage_id = storage_id_for_user_id(user_id)
+
+    FirehoseClient.instance.put_record(
+      :analysis,
+      {
+        study: 'reset-progress',
+        event: 'admin-delete-progress',
+        user_id: user_id,
+        script_id: script_id,
+        data_json: {
+          signed_in_user: current_user.username,
+          reason: params[:reason]
+        }.to_json
+      }
+    )
+
+    UserScript.where(user_id: user_id, script_id: script_id).destroy_all
+    UserLevel.where(user_id: user_id, script_id: script_id).destroy_all
+    ChannelToken.where(storage_id: user_storage_id, script_id: script_id).destroy_all unless user_storage_id.nil?
+    TeacherFeedback.where(student_id: user_id, script_id: script_id).destroy_all
+  end
```

Comment:
Do we also need to delete from Level Sources?

Reply:
It's a bit tricky to delete from `level_sources` because multiple `user_level` rows can point to the same level_sources row.  I put in the dev spec that we're just going to leave the level_sources rows.